### ISSUE TITLE
Wrong diagnostics: error: _Atomic cannot be applied to incomplete type

### ISSUE THREAD

**Issue by pmor13** [llvm/llvm-project#54538]
```
_Atomic void* p;
_Atomic struct S* p1;
_Atomic struct S x;
struct S { int i; };
```

```
$ clang t0.c -std=c11 -pedantic -Wall -Wextra -c
<source>:1:1: error: _Atomic cannot be applied to incomplete type 'void' 
<source>:2:1: error: _Atomic cannot be applied to incomplete type 'struct S' 
<source>:3:1: error: _Atomic cannot be applied to incomplete type 'struct S'
```
In the C standard (all revisions) there is nothing, which forbids / restricts applying `_Atomic` to an incomplete type.

**Comment by frederick-vs-ja** [comment]
Duplicate of #48742.

**Comment by frederick-vs-ja** [comment]
Duplicate of #36585.
